components/layout_2020/flow/construct.rs

```diff
@@ -493,7 +494,9 @@ where
         let kind = match contents {
             Contents::NonReplaced(contents) => match display_inside {
                 DisplayInside::Flow { is_list_item }
-                    if !info.style.establishes_block_formatting_context() =>
+                    if !info.style.establishes_block_formatting_context(
+                        BaseFragmentInfo::anonymous().flags,
```

Comment:
It's fine since `effective_overflow` only checks for replacedness and this is not replaced. But this isn't passing the actual flags, e.g. this could be the `<body>` but the `IS_BODY_ELEMENT_OF_HTML_ELEMENT_ROOT` flag would be missing.

Seems worth adding a comment. Also,

```suggestion
                        FragmentFlags::empty(),
```